umbral/utils.py

```diff
@@ -55,6 +55,42 @@ def hash_to_bn(list, params):
  
     return res
 
+def hash_to_point(curve, data, constant=None):
+    """
+    Hashes arbitrary data into a valid EC point of the specified curve,
+    using the try-and-increment method.
+    It admits an optional constant as an additional input to the hash function.
+    It uses SHA256 as the internal hash function. 
+
+    WARNING: Do not use when the input data is secret, as this implementation is not 
+    in constant time, and hence, it is not safe with respect to timing attacks.
+
+    TODO: Check how to uniformly generate ycoords. Currently, it only outputs points 
+    where ycoord is even (i.e., starting with 0x02 in compressed notation)
+    """
+    if constant is None:
+        constant = []
+
+    # We use a 32-bit counter as additional input
+    i = 1
+    while i < 2**32:
+        ibytes = i.to_bytes(4, byteorder='big')
+        digest = hashes.Hash(hashes.SHA256(), backend=default_backend())
+        digest.update(constant + ibytes + data)
+        hash = digest.finalize()
+
+        compressed02 = b"\x02"+hash
+        try:
+            h = Point.from_bytes(compressed02, curve)
+            return h
+        except:
+            pass
+
+        i+=1
+
+    # Only happens with probability 2^(-32)
+    raise ValueError('Could not find %c in %s' % (ch,str))
```

Comment:
Let's do the Python3 specific method for string formatting:

`raise ValueError('Could not find {} in {}'.format(ch, str))`

Also, I don't see `ch` or `str` defined in this part. What's going on here?